pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -0,0 +1,141 @@
+		switch err {
+		case util.ErrInvalidBucketAccessClass:
+			glog.V(5).Infof("Bucket access class specified does not exist. Stop provisioning, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.UID)
+			err = nil
+		case util.ErrBucketAccessAlreadyExists:
+			glog.V(5).Infof("Bucket access already exist for this bucket request. Stop provisioning, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.UID)
+			err = nil
+		default:
+			glog.V(2).Infof("Final error received, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.Name)
+		}
+		return err
+	}
+
+	glog.V(5).Infof("BucketAccessRequest processing succeeded, removing bucket access request %s from bucket access requests in progress", bucketAccessRequest.Name)
+	return nil
+}
+
+func (b *bucketAccessRequestListener) Update(ctx context.Context, old, new *v1alpha1.BucketAccessRequest) error {
+	glog.V(1).Infof("update called for bucket %v", old)
+	return nil
+}
+
+func (b *bucketAccessRequestListener) Delete(ctx context.Context, obj *v1alpha1.BucketAccessRequest) error {
+	glog.V(1).Infof("delete called for bucket %v", obj)
+	return nil
+}
+
+// provisionBucketAccess  attempts to provision a BucketAccess for the given bucketAccessRequest.
+// Returns nil error only when the bucketaccess was provisioned. An error is return if we cannot create bucket access.
+// A normal error is returned when  bucket acess  was not provisioned and provisioning should be retried (requeue the bucketAccessRequest),
+// or a special error  errBucketAccessAlreadyExists, errInvalidBucketAccessClass is returned when provisioning was impossible and
+// no further attempts to provision should be tried.
+func (b *bucketAccessRequestListener) provisionBucketAccess(ctx context.Context, bucketAccessRequest *v1alpha1.BucketAccessRequest) error {
+	bucketAccessClassName := bucketAccessRequest.Spec.BucketAccessClassName
+
+	bucketaccess := b.FindBucketAccess(ctx, bucketAccessRequest)
+	if bucketaccess != nil {
+		// bucketaccess has been already provisioned, nothing to do.
+		return util.ErrBucketAccessAlreadyExists
+	}
+
+	bucketAccessClass, err := b.bucketClient.ObjectstorageV1alpha1().BucketAccessClasses().Get(ctx, bucketAccessClassName, metav1.GetOptions{})
+	if bucketAccessClass == nil {
+		// bucket has been already provisioned, nothing to do.
+		return util.ErrInvalidBucketAccessClass
+	}
+
+	bucketRequest, err := b.bucketClient.ObjectstorageV1alpha1().BucketRequests(bucketAccessRequest.Namespace).Get(ctx, bucketAccessRequest.Spec.BucketRequestName, metav1.GetOptions{})
+	if bucketRequest == nil {
+		// bucket request does not exist, we have to reject this provision.
+		return util.ErrInvalidBucketRequest
+	}
+	if err != nil {
+		return err
+	}
+
+	if bucketRequest.Spec.BucketInstanceName == "" {
+		return util.ErrWaitForBucketProvisioning
```

Comment:
I see BucketInstanceName filled in below, but how do we ever get there? Does the sidecar fill in this name?